src/client/Microsoft.Identity.Client.Broker/WamAdapters.cs

```diff
@@ -277,7 +277,7 @@ private static void AddPopParams(AuthenticationRequestParameters authenticationR
         {
             MsalTokenResponse msalTokenResponse = null;
 
-            if (authResult.IsSuccess)
+            if (TokenReceivedFromWam(authResult, logger))
             {
                 msalTokenResponse = ParseRuntimeResponse(authResult, authenticationRequestParameters, logger);
```

Comment:
nit: `ParseRuntimeResponse` and `ThrowExceptionFromWamError` can be made private, so that we know to use `HandleResponse` outside of this class.